Ical.Net/Utility/TextUtil.cs

```diff
@@ -4,32 +4,73 @@
 //
 
 using System;
+using System.Buffers;
 using System.Collections.Generic;
 using System.IO;
+using System.Text;
 using System.Text.RegularExpressions;
 using Ical.Net.Serialization;
 
 namespace Ical.Net.Utility;
 
 internal static class TextUtil
 {
-    /// <summary> Folds lines at 75 characters, and prepends the next line with a space per RFC https://tools.ietf.org/html/rfc5545#section-3.1 </summary>
-    public static string FoldLines(string incoming)
+    /// <summary>
+    /// Folds lines to 75 octets. It appends a CrLf, and prepends the next line with a space.
+    /// per RFC https://tools.ietf.org/html/rfc5545#section-3.1
+    /// </summary>
+    public static void FoldLines(this StringBuilder result, string inputLine)
     {
-        //The spec says nothing about trimming, but it seems reasonable...
-        var trimmed = incoming.Trim();
-        if (trimmed.Length <= 75)
+        if (Encoding.UTF8.GetByteCount(inputLine) <= 75)
         {
-            return trimmed + SerializationConstants.LineBreak;
+            result.Append(inputLine);
+            result.Append(SerializationConstants.LineBreak);
+            return;
         }
 
-        const int takeLimit = 74;
+        // Use ArrayPool<char> to rent arrays for processing
+        // Cannot use Span<char> and stackalloc because netstandard2.0 does not support it
+        var arrayPool = ArrayPool<char>.Shared;
```

Comment:
I see the performance optimization the usage of the ArrayPool could bring but I would usually try to avoid using a global pool here. It increases complexity and could have unexpected side effects. Had similar situations (in other languages though) where some memory sanitizer suddenly complained that the globally allocated memory changed after calling a certain method just because of some global cache. This might not be the case here but given the quite small effect we can expect from the optimization (compared to just allocating a new array) and given the fact that we don't know how/where users of the lib will will use it, I would rather avoid it.

Question is, whether its needed at all. Why not directly append to the result?

Reply:
KISS, replaced it